sdk/schemaregistry/azure-data-schemaregistry-avro/src/main/java/com/azure/data/schemaregistry/avro/SchemaRegistryAvroAsyncDeserializer.java

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.data.schemaregistry.avro;
+
+import com.azure.data.schemaregistry.SerializationException;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Scheduler;
+import reactor.core.scheduler.Schedulers;
+
+import java.util.concurrent.Executors;
+
+/**
+ * Asynchronous registry-based deserializer implementation.
+ */
+public class SchemaRegistryAvroAsyncDeserializer {
+    private static final int DEFAULT_THREAD_POOL_SIZE = 8;
+
+    private final SchemaRegistryAvroDeserializer deserializer;
+    private final Scheduler scheduler;
```

Comment:
Is there a specific reason we are creating a custom scheduler for deserialization?

Reply:
Async implementation is using a threadpool to do sync work, hence the scheduler with the dedicated threadpool

Reply:
This will be changed for GA.